lib/recipe.rb

```diff
@@ -98,7 +98,12 @@
     solaris2
     suse
   )           => 'root',
-  %i(windows) => 'administrator'
+  %i(windows) =>  if platform?('windows')
+                    # This class is only loaded on Windows
```

Comment:
Out of curiosity, does the constant Chef::ReservedNames::Win32::Security::SID get evaluated if value_for_platform_family isn't windows? I would have thought that the platform? call wasn't needed, but always happy to learn.

Reply:
Yeah. I was disappointed it couldn't be that simple.

```
btm@oak:~$ /opt/chef/embedded/bin/irb
irb(main):001:0> require 'chef'
=> true
irb(main):002:0> admin_user = value_for_platform_family(
irb(main):003:1*   %i(
irb(main):004:1]     aix
irb(main):005:1]     amazon
irb(main):006:1]     debian
irb(main):007:1]     fedora
irb(main):008:1]     freebsd
irb(main):009:1]     mac_os_x
irb(main):010:1]     oracle
irb(main):011:1]     rhel
irb(main):012:1]     solaris2
irb(main):013:1]     suse
irb(main):014:1]   )           => 'root',
irb(main):015:1*   %i(windows) => Chef::ReservedNames::Win32::Security::SID.admin_account_name
irb(main):016:1> )
NameError: uninitialized constant Chef::ReservedNames::Win32::Security
Did you mean?  SecurityError
        from (irb):15
        from /opt/chef/embedded/bin/irb:11:in `<main>'
```

Reply:
simpler example:
```
chef:recipe (13.0.118)> platform?('windows')
 => false
chef:recipe (13.0.118)> admin_user = value_for_platform_family('windows' => Chef::ReservedNames::Win32::Security::SID.admin_account_name)
NameError: uninitialized constant Chef::ReservedNames::Win32::Security
        from (irb#1):56
```